### PR TITLE
Add Microsoft Teams notifier

### PR DESCRIPTION
**Pull request by phaelin**
I also cleaned up some formatting on the README and added the Jetbrains `.idea` folder to the `.gitignore`.

### CODE REVIEW

**Comment by phaelin** [comment]
@smartinez87 I don't often contribute to open source projects, but this custom notifier has been a major improvement for my team, so I thought I'd add it to the gem so others could benefit from it.

Teams expects incoming webhooks to follow a standard format, so this notifier ensures all of that is handled behind the scenes. Here is a sample screenshot of the end-result, with proprietary info masked:

image

File: lib/exception_notifier/teams_notifier.rb [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    end
+
+    def details
+      details = {
+        "title" => "Details",
+        "facts" => []
+      }
+
+      details['facts'].push message_request unless @request_items.nil?
+      details['ß'].push message_backtrace unless @backtrace.nil?
```
Comment (aried3r) [inline lib/exception_notifier/teams_notifier.rb]: What is this being used for? I couldn't find anything in the Microsoft Teams [cards documentation](https://docs.microsoft.com/en-us/microsoftteams/platform/concepts/cards/cards-reference) regarding a `ß` parameter.

File: README.md [hunk 2]
```diff
@@ -794,6 +795,60 @@ Number of backtrace lines to be displayed in the notification message. By defaul
...
+  }
+```
+
+#### Options
+
+##### webhook_url
+
+*String, required*
+
+The Incoming WebHook URL on mattermost.
```
Comment (FLarra) [inline README.md]: It seems like in the copy&paste you forgot to change the integration name. Can you please update it later?

Comment (phaelin) [reply README.md]: Absolutely. Thanks!

**Pull request merged**